registry_provider.go

```diff
@@ -0,0 +1,256 @@
+const (
+	PrivateRegistry RegistryName = "private"
+	PublicRegistry  RegistryName = "public"
+)
+
+// RegistryProviderIncludeOps represents which jsonapi include can be used with registry providers
+type RegistryProviderIncludeOps string
+
+// List of available includes
+const (
+	RegistryProviderVersionsInclude RegistryProviderIncludeOps = "registry-provider-versions"
+)
+
+// RegistryProvider represents a registry provider
+type RegistryProvider struct {
+	ID           string                      `jsonapi:"primary,registry-providers"`
+	Name         string                      `jsonapi:"attr,name"`
+	Namespace    string                      `jsonapi:"attr,namespace"`
+	CreatedAt    string                      `jsonapi:"attr,created-at,iso8601"`
+	UpdatedAt    string                      `jsonapi:"attr,updated-at,iso8601"`
+	RegistryName RegistryName                `jsonapi:"attr,registry-name"`
+	Permissions  RegistryProviderPermissions `jsonapi:"attr,permissions"`
+
+	// Relations
+	Organization             *Organization              `jsonapi:"relation,organization"`
+	RegistryProviderVersions []*RegistryProviderVersion `jsonapi:"relation,registry-provider-versions"`
+
+	// Links
+	Links map[string]interface{} `jsonapi:"links,omitempty"`
+}
+
+type RegistryProviderPermissions struct {
+	CanDelete      bool `jsonapi:"attr,can-delete,omitempty"`
+	CanUploadAsset bool `jsonapi:"attr,can-upload-asset,omitempty"`
+}
+
+type RegistryProviderListOptions struct {
+	ListOptions
+
+	// Optional: A query string to filter by registry_name
+	RegistryName RegistryName `url:"filter[registry_name],omitempty"`
+
+	// Optional: A query string to filter by organization
+	OrganizationName string `url:"filter[organization_name],omitempty"`
+
+	// Optional: A query string to do a fuzzy search
+	Search string `url:"q,omitempty"`
+
+	// Optional: Include related jsonapi relationships
+	Include *[]RegistryProviderIncludeOps `url:"include,omitempty"`
+}
+
+type RegistryProviderList struct {
+	*Pagination
+	Items []*RegistryProvider
+}
+
+// RegistryProviderID is the multi key ID for addressing a provider
+type RegistryProviderID struct {
+	OrganizationName string
+	RegistryName     RegistryName
+	Namespace        string
+	Name             string
+}
+
+// RegistryProviderCreateOptions is used when creating a registry provider
+type RegistryProviderCreateOptions struct {
+	// Type is a public field utilized by JSON:API to
+	// set the resource type via the field tag.
+	// It is not a user-defined value and does not need to be set.
+	// https://jsonapi.org/format/#crud-creating
+	Type string `jsonapi:"primary,registry-providers"`
+
+	Name         string       `jsonapi:"attr,name"`
+	Namespace    string       `jsonapi:"attr,namespace"`
+	RegistryName RegistryName `jsonapi:"attr,registry-name"`
+}
+
+type RegistryProviderReadOptions struct {
+	// Optional: Include related jsonapi relationships
+	Include *[]RegistryProviderIncludeOps `url:"include,omitempty"`
```

Comment:
The empty value of a slice is nil, which makes pointers to slices a little symbolically redundant and harder to use. I suggest removing the pointer from the typedef